### PR TITLE
Development guide doc for C# & Xamarin

### PR DESCRIPTION
**Pull request by ravinderjangra**
Example project code is in the review so the link in Articles is not available yet. Please use following links for example code.
- Desktop/Console app branch link: https://github.com/maidsafe/safe-getting-started-dotnet/

### CODE REVIEW

File: src/contents/en-gb/doc_dotnet.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+```csharp
+var encodedReq = await Session.EncodeAuthReqAsync(authReq);
+var url = $"safe-auth://{encodedReq}";
+System.Diagnostics.Process.Start(url);
+```
+**6. Grant access**
+
+Once the authorisation request is received the Authenticator launches and a pop-up dialogue prompts for access (Allow or Deny). The Authenticator then sends the response back to the application.
+
+Once the authorisation request is received, the Authenticator sends an authorisation response using the system URI mechanism. The system matches the URI scheme to the application and launches the application passing the URI as an argument.
```
Comment (S-Coyle) [inline src/contents/en-gb/doc_dotnet.md]: The beginning of this paragraph is almost identical to the previous paragraph which is a little confusing - would it make sense to add in the word "confirmation" to differentiate?
"Once the authorisation request confirmation is received..."

File: src/contents/en-gb/doc_dotnet.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+## Create mutable data
+
+Mutable data is a key-value store which can be created at either a specific address on the network or at a random address. It can be publicly available (public mutable data) or encrypted (private mutable data). Mutable data has a type tag associated with it (a number) that can be chosen while creating the mutable data.
+
+To learn more about the mutable Data and other data types in SAFE Network visit [Discover Page](/discover).
+
+For this tutorial, we will: 
+- Create a mutable data object in the network to store the application data.
+- Create public mutable data at a random address (this address represents a unique 256 bit address in the network space. Learn more about XOR addresses at [MaidSafe's blog](https://blog.maidsafe.net/2016/05/27/structuring-networks-with-xor)).
```
[the remark this answers is not in the document]
Comment (S-Coyle) [reply src/contents/en-gb/doc_dotnet.md]: Update - new article published here https://medium.com/@maidsafe/structuring-network-with-xor-431e785b5ee7
Please update to point to this link (blog post fwd does not work)

File: src/contents/en-gb/doc_dotnet.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+For both update and remove (not insert) a 1 is added to `value.Item2` and passed to the entry action function. This allows for versioning.
+
+## Versioning in mutable data
+
+Each entry in a mutable data has a version associated to it - a numeric value. 
+
+When a new entry is inserted it's inserted with version 0. Every time a mutation is performed this version increases by one, as specified in the above code. This is used by the network to ensure only one mutation is applied when simultaneous mutation requests are received for the same version of an entry. This ensures the state change of such an entry is what the originator of the request is intending to do.
+
+When an entry is removed it is never deleted from the mutable data, only the value is cleared. This ensures a new entry with the same key cannot be inserted - rather it is updated (ravi: does this sentence read correctly? I think I've understood what you meant but please double check).
```
Comment (S-Coyle) [inline src/contents/en-gb/doc_dotnet.md]: Looks like there's a comment from your notes still in at the end of this sentence

File: src/contents/en-gb/doc_dotnet.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravinderjangra) [reply src/contents/en-gb/doc_dotnet.md]: Will update the doc once the content is finalized on the HackMD.

File: src/contents/en-gb/doc_dotnet.md (hunk 1 above)
Comment (ravinderjangra) [reply src/contents/en-gb/doc_dotnet.md]: Will update the doc once the content is finalized on the HackMD.

File: src/contents/en-gb/doc_dotnet.md (hunk 3 above)
Comment (ravinderjangra) [reply src/contents/en-gb/doc_dotnet.md]: Will update the doc once the content is finalized on the HackMD.

**Pull request merged**